operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java

```diff
@@ -391,4 +391,14 @@ default boolean parseResourceVersionsForEventFilteringAndCaching() {
     return false;
   }
 
+  /**
+   * {@link io.javaoperatorsdk.operator.api.reconciler.UpdateControl} patchStatus can either use
+   * simple update or SSA for status subresource patching.
+   *
+   * @return true by default
+   */
+  default boolean useSSAForResourceStatusPatch() {
```

Comment:
Will add an additional test for upgradint from non-SSA to SSA, just added for safety reasons: since there were issues with SSA in the past, also even the non-built in kubernetes controller are not using SSA. So it does not harm to add a feature flag.

Reply:
Well, it adds complexity for not much. I'd rather we only add the flag if we determine it's needed.

Reply:
I think the migration from non-SSA to SSA - is not trivial. At least I hit some strange issues today, but will add test. Especially actual users would be safer to have this flag.